utilities/tools/Test-NamePrefixAvailability.ps1

```diff
@@ -56,123 +63,139 @@ function Test-NamePrefixAvailability {
-
-        # Replace parameter file tokens
-        # -----------------------------
-
-        # Tokens in settings.yml
-        $GlobalVariablesObject = Get-Content -Path (Join-Path $repoRoot 'settings.yml') | ConvertFrom-Yaml -ErrorAction Stop | Select-Object -ExpandProperty variables
-
-        # Construct Token Configuration Input
-        $tokenConfiguration = @{
-            FilePathList = $parameterFiles
-            Tokens       = @{
-                'namePrefix' = $namePrefix
-            }
-            TokenPrefix  = $GlobalVariablesObject | Select-Object -ExpandProperty tokenPrefix
-            TokenSuffix  = $GlobalVariablesObject | Select-Object -ExpandProperty tokenSuffix
-        }
-
-        # Add additional tokens provided by the user
-        $tokenConfiguration.Tokens += $Tokens
-
-        # Invoke Token Replacement Functionality and Convert Tokens in Parameter Files
-        $null = Convert-TokensInFileList @tokenConfiguration
-
-        try {
 
-            # Extract Parameter Names
-            # -----------------------
-            $storageAccountFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.Storage/storageAccounts' }
-            $storageAccountNames = $storageAccountFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
+        $storageAccountNames = @()
+        $containerRegistryNames = @()
+        $keyVaultNames = @()
 
-            $keyVaultFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.KeyVault/vaults' }
-            $keyVaultNames = $keyVaultFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
+        $parameterFiles = (Get-ChildItem -Path $repoRoot -Recurse -Filter 'deploy.test.bicep').FullName | ForEach-Object { $_.Replace('\', '/') }
 
-            $acrFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.ContainerRegistry/registries' }
-            $acrNames = $acrFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
-
-            $subscriptionId = (Get-AzContext).Subscription.Id
-
-            # Storage
-            Write-Host "`nCheck Storage Accounts" -ForegroundColor 'Cyan'
-            Write-Host '======================' -ForegroundColor 'Cyan'
-            foreach ($storageAccountName in $storageAccountNames) {
-                $path = '/subscriptions/{0}/providers/Microsoft.Storage/checkNameAvailability?api-version=2021-04-01' -f $subscriptionId
-                $requestInputObject = @{
-                    Method  = 'POST'
-                    Path    = $path
-                    Payload = @{
-                        name = $storageAccountName
-                        type = 'Microsoft.Storage/storageAccounts'
-                    } | ConvertTo-Json
+        foreach ($relevantResourceType in $relevantResourceTypes) {
```

Comment:
The loops should be switched for an improved runtime. I.e.:
1. Outer Loop: Go through the parameter files
2. Inner Loop: Go through the Resource Types
Reason being that you then loop infinitely less through the big nummber of parameter files over and over again (and read their content)

Reply:
not sure I get your point correctly, but I tried switching the inner with the outer loop.

However the performance difference should be neglectable, right? The API calls are what take up 99% of the runtime of the script.

Reply:
Doesn't mean we shouldn't code by example ;)

Reply:
true :D